beacon_node/src/config.rs

```diff
@@ -270,14 +271,29 @@ pub fn get_configs<E: EthSpec>(
      * Zero-ports
      *
      * Replaces previously set flags.
+     * Libp2p and discovery ports are set explicitly by selecting
+     * a random free port so that we aren't needlessly updating ENR
+     * from lighthouse.
+     * Discovery address is set to localhost by default.
      */
     if cli_args.is_present("zero-ports") {
-        client_config.network.libp2p_port = 0;
-        client_config.network.discovery_port = 0;
+        if client_config.network.discovery_address == IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)) {
+            client_config.network.discovery_address = "127.0.0.1"
+                .parse()
+                .map_err(|e| format!("Invalid discovery address: {}", e))?
+        }
+        client_config.network.libp2p_port =
+            unused_port("tcp").map_err(|e| format!("Failed to get port for libp2p: {}", e))?;
+        client_config.network.discovery_port =
+            unused_port("udp").map_err(|e| format!("Failed to get port for discovery: {}", e))?;
         client_config.rest_api.port = 0;
         client_config.websocket_server.port = 0;
     }
 
+    // ENR ip needs to be explicit for node to be discoverable
```

Comment:
@AgeManning This check is restrictive as we have to explicitly pass the `discovery-address` even for nodes which aren't bootnodes. 

Non-bootnodes will work without this check as their ENR will get updated after dialing the bootnode. 
I can't think of a good way to add this check just for bootnodes.

Reply:
I think we can just set it to "127.0.0.1" as you've done before. But perhaps log a warning saying that we have done this.